cdap-app-fabric/src/main/java/co/cask/cdap/scheduler/NotificationSubscriberService.java

```diff
@@ -319,15 +334,31 @@ protected void updateJobQueue(DatasetContext context, Notification notification)
   }
 
   private static final class Job {
-    ProgramSchedule schedule;
+    private final ProgramSchedule schedule;
+    private final Map<String, String> systemArgs;
```

Comment:
I am not sure why they are needed here. Can't we fetch that from the schedule when the job is launched? Or is the concern that the schedule may be updated/changed/deleted before the job is launched?